packages/ag-charts-enterprise/src/features/annotations/annotations.ts

```diff
@@ -119,9 +119,28 @@ class AxesButtons {
 }
 
 export class Annotations extends _ModuleSupport.BaseModuleInstance implements _ModuleSupport.ModuleInstance {
-    @_ModuleSupport.ObserveChanges<Annotations>((target, enabled) => {
-        target.ctx.toolbarManager.toggleGroup('annotations', 'annotations', Boolean(enabled));
-        if (!enabled) target.clear();
+    // TODO: When the 'restore-annotations' event is triggered from `ActionsOnSet.newValue()`, the module is still
+    // disabled when `onRestoreAnnotations()` is called, preventing the state from being restored. However,
+    // when `ObserveChanges()` is first called `target.enabled === false`, rather than `undefined`. So
+    // there is no way to detect if the module was actively disabled. This flag simulates a combined
+    // behaviour of both and is toggled when the module is actively disabled and enabled.
+    private __hackWasDisabled = false;
```

Comment:
I'll aim to remove this hack soon, but I do not wish to mess with these decorators close to release.